File: Number Operation/prg.dart
import 'dart:io';

void main() {
  int a = int.parse(stdin.readLineSync()!);
  int b = int.parse(stdin.readLineSync()!);

  print("AND: ${a & b}");
  print("OR: ${a | b}");
  print("XOR: ${a ^ b}");
  print("NOT A: ${~a}");
  print("Left Shift: ${a << 1}");
  print("Right Shift: ${a >> 1}");

  bool x = stdin.readLineSync()!.toLowerCase() == 'true';
  bool y = stdin.readLineSync()!.toLowerCase() == 'true';
  print("Logical AND: ${x && y}");
  print("Logical OR: ${x || y}");
  print("Logical NOT X: ${!x}");
}

File: Number Operation/prg.go
package main

import "fmt"

func main() {
    var a, b int
    fmt.Scan(&a, &b)

    fmt.Println("AND:", a&b)
    fmt.Println("OR:", a|b)
    fmt.Println("XOR:", a^b)
    fmt.Println("NOT A:", ^a)
    fmt.Println("Left Shift:", a<<1)
    fmt.Println("Right Shift:", a>>1)

    var x, y bool
    fmt.Scan(&x, &y)
    fmt.Println("Logical AND:", x && y)
    fmt.Println("Logical OR:", x || y)
    fmt.Println("Logical NOT X:", !x)
}
